src/entities/job.ts

```diff
@@ -25,48 +25,48 @@ export type Payload = {
   branchName: string;
   isFork: boolean;
   private: boolean;
-  isXlarge: boolean | null | undefined;
+  isXlarge?: boolean | null;
   repoOwner: string;
   url: string;
-  newHead: string | null | undefined;
-  patch: string | null | undefined;
-  alias: string | null | undefined;
-  manifestPrefix: string | undefined;
-  pathPrefix: string | null | undefined;
-  mutPrefix: string | null | undefined;
-  aliased: boolean | null | undefined;
-  primaryAlias: string | null | undefined;
-  repoBranches: any;
-  stable: boolean | null | undefined;
-  isNextGen: boolean | null | undefined;
-  regression: boolean | null | undefined;
-  urlSlug: string | null | undefined;
+  newHead?: string | null;
+  patch?: string | null;
+  alias?: string | null;
+  manifestPrefix?: string;
+  pathPrefix?: string | null;
+  mutPrefix?: string | null;
+  aliased?: boolean | null;
+  primaryAlias?: string | null;
+  repoBranches?: any;
+  stable?: boolean | null;
+  isNextGen?: boolean | null;
+  regression?: boolean | null;
+  urlSlug?: string | null;
   prefix: string;
   project: string;
-  includeInGlobalSearch: boolean;
+  includeInGlobalSearch?: boolean;
 };
 
 export type Job = {
   _id: string;
   payload: Payload;
   createdTime: Date;
-  endTime: Date | null | undefined;
-  error: any | null | undefined;
-  logs: string[] | null | undefined;
-  priority: number | null | undefined;
-  result: any | null | undefined;
-  startTime: Date;
+  endTime?: Date | null;
+  error?: any;
+  logs?: string[] | null;
+  priority?: number;
+  result?: any;
+  startTime: Date | null;
```

Comment:
I made this null because in the `prepGithubPushPayload` function, we assign null to this value. This then gets upserted into the collection, so it appears that we have been providing a null value for this one previously.